runtime-modules/content/src/nft/types.rs

```diff
@@ -4,7 +4,7 @@ use super::*;
 pub type NftMetadata = Vec<u8>;
 
 /// Owner royalty
-pub type Royalty = Perbill;
+pub type Royalty<AccountId> = (Perbill, AccountId);
```

Comment:
Yes, this is not correct. We do not want the issuer to lock in an account, we only want the `complete_auction` method to statically either get a bompined royalty % & account, or neither. So you should just have left this type alone and updates signature of `complete_payment` to be `(..., royalty_payment: Option<Royalty, AccountId>`.